plugins/inventory/d42.py

```diff
@@ -115,3 +114,537 @@ def parse(self, inventory, loader, path, cache=False):
+            view_vendor_v1.name as manufacturer,
+            view_device_v1.type as hw_model,
+            view_hardware_v1.hardware_pk as hw_model_id,
+            vendor.name as manufacturer,
+            view_hardware_v1.is_it_switch,
+            view_hardware_v1.is_it_blade_host,
+            view_hardware_v1.depth as hw_depth,
+            view_hardware_v1.size as hw_size,
+            vhost.name as virtual_host_name,
+            nonauthoritativealias.alias_names as nonauthoritativealiases,
+            alias.alias_names as aliases,
+            obj_category.name as category,
+            case
+                WHEN view_device_v1.hz = 'MHz' and view_device_v1.cpupower is not null
+                    THEN view_device_v1.cpupower / 1000
+                ELSE
+                    view_device_v1.cpupower
+                END as cpuspeed,
+            case
+                WHEN view_device_v1.ram_size_type = 'MB' and view_device_v1.ram is not null
+                    THEN view_device_v1.ram / 1024
+                WHEN view_device_v1.ram_size_type = 'TB' and view_device_v1.ram is not null
+                    THEN view_device_v1.ram * 1024
+                ELSE
+                    view_device_v1.ram
+                END as ramsize,
+            network_ip.ip_address, network_hw.hwaddress,
+            CEIL(COALESCE(p.total_part_disk_size,
+                            CASE
+                                WHEN view_device_v1.hard_disk_count IS NOT NULL AND
+                                     view_device_v1.hard_disk_size IS NOT NULL AND
+                                     view_device_v1.hard_disk_size_type IS NOT NULL THEN
+                                        view_device_v1.hard_disk_count *
+                                        CASE
+                                            WHEN view_device_v1.hard_disk_size_type = 'GB'
+                                                THEN view_device_v1.hard_disk_size
+                                            WHEN view_device_v1.hard_disk_size_type = 'TB'
+                                                THEN view_device_v1.hard_disk_size * 1024
+                                            WHEN view_device_v1.hard_disk_size_type = 'PB'
+                                                THEN view_device_v1.hard_disk_size * 1024 * 1024
+                                            ELSE
+                                                NULL
+                                        END
+                                ELSE NULL
+                            END)) AS disk_size
+            from view_device_v1
+            left join view_hardware_v1
+                on view_device_v1.hardware_fk = view_hardware_v1.hardware_pk
+            left join  (select view_ipaddress_v1.device_fk,  string_agg(host(view_ipaddress_v1.ip_address)::character varying, ', ' order by view_ipaddress_v1.ip_address) as ip_address from view_ipaddress_v1 group by view_ipaddress_v1.device_fk) network_ip
```

Comment:
Are the joins to view_ipaddress_v1, view_netport_v1, view_purchaselineitem_v1 still needed since separate queries are being used to get them?

Reply:
> I made all the queries perform a join strictly for device name. the name is not used when creating the json, but I wanted the queries to be reusable and be useful for debugging. if for any reason a device is not being imported support can call the same query and find what device information is not coming in for what device